PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = ['event.type.mail']
-    _inherit = ['event.mail']
-    _inherit = ['event.mail.registration']
-    _inherit = ['sms.template']

Total Changes: 8

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/event_sms/models/event_mail.py
+++ b/addons/event_sms/models/event_mail.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class EventTypeMail(models.Model):
-    _inherit = ['event.type.mail']
+    _inherit = 'event.type.mail'
 
     notification_type = fields.Selection(selection_add=[('sms', 'SMS')])
     template_ref = fields.Reference(ondelete={'sms.template': 'cascade'}, selection_add=[('sms.template', 'SMS')])
@@ -17,7 +17,7 @@ class EventTypeMail(models.Model):
 
 
 class EventMail(models.Model):
-    _inherit = ['event.mail']
+    _inherit = 'event.mail'
 
     notification_type = fields.Selection(selection_add=[('sms', 'SMS')])
     template_ref = fields.Reference(ondelete={'sms.template': 'cascade'}, selection_add=[('sms.template', 'SMS')])

--- a/addons/event_sms/models/event_mail_registration.py
+++ b/addons/event_sms/models/event_mail_registration.py
@@ -2,7 +2,7 @@ from odoo import fields, models
 
 
 class EventMailRegistration(models.Model):
-    _inherit = ['event.mail.registration']
+    _inherit = 'event.mail.registration'
 
     def _execute_on_registrations(self):
         todo = self.filtered(

--- a/addons/event_sms/models/sms_template.py
+++ b/addons/event_sms/models/sms_template.py
@@ -6,7 +6,7 @@ from odoo.osv import expression
 
 
 class SmsTemplate(models.Model):
-    _inherit = ['sms.template']
+    _inherit = 'sms.template'
 
     @api.model
     def _search_display_name(self, operator, value):
